v2/concordium/types.proto

```diff
@@ -2168,6 +2168,205 @@ message ElectionInfo {
+      message NetworkStats {                
+        // The number of messages sent to the peer.
+        uint64 packets_sent = 2;
+    
+        // The number of messages received from the peer.
+        uint64 packets_received = 3;
+    
+        // The connection latency (i.e., ping time) in milliseconds.
+        uint64 latency = 4;
+      }
+
+      // The identifier of the peer that it
+      // wishes to be identified by.
+      PeerId peer_id = 1;
+      // The port of the peer.
+      IpSocketAddress socket_address = 2;
+      // Network related statistics for the peer.
+      NetworkStats network_stats = 3;
+      // consensus related information of the peer.
+      oneof consensus_info {
+        // The peer is of type `Bootstrapper` is not participating in consensus 
+        // and thus has no catchup status.
+        Empty bootstrapper = 4;
+        // The peer is a regular node and have 
+        // an associated catchup status.
+        CatchupStatus node_catchup_status = 5;
+      }
+  }
+
+  repeated Peer peers = 1;
+}
+
+// Node info response
+// Contains various information of the
+// enquired node.
+message NodeInfo {
+  // Network related information of the node.
+  message NetworkInfo {   
+    // The node id.
+    PeerId node_id = 1;
+    // Total number of packets sent by the node.
+    uint64 peer_total_sent = 2;
+    // Total number of packets received by the node.
+    uint64 peer_total_received = 3;
+    // Average outbound throughput in bytes per second.
+    uint64 avg_bps_in = 4;
+    // Average inbound throughput in bytes per second.
+    uint64 avg_bps_out = 5;
+  }
+
+  // Consensus info for a node configured with baker keys.
+  message BakerConsensusInfo {
+    // The committee information of a node configured with 
+    // baker keys but somehow the node is part of the 
+    // baking committee.
+    enum PassiveCommitteeInfo {
+      // The node is started with baker keys however it is currently not in the baking committee.
+      // The node is __not__ baking.
+      NOT_IN_COMMITTEE = 0;
+      // The account is registered as a baker but not in the current `Epoch`.
+      // The node is __not__ baking.
+      ADDED_BUT_NOT_ACTIVE_IN_COMMITTEE = 1;
+      // The node has configured invalid baker keys i.e., the configured
+      // baker keys do not match the current keys on the baker account.
+      // The node is __not__ baking.
+      ADDED_BUT_WRONG_KEYS = 2;
+    }
+
+    // The node is configured with baker keys and 
+    // is member of the baking committee.
+    message ActiveBakerCommitteeInfo {
+      BakerId baker_id = 1;
+    }
+
+    // The node is configured with baker keys and 
+    // is member of the baking and finalization committees.
+    message ActiveFinalizerCommitteeInfo {
+      BakerId baker_id = 1;
+    }
+  
+    // Status of the baker configured node.
+    oneof status {
+      // The node is currently not baking.
+      PassiveCommitteeInfo passive_committee_info = 2;
```

Comment:
So either just add a `baker_id` to this entire `BakerConsensusInfo` struct, and then change the `ActiveBaker...` to be equivalent to `Empty`, or add `baker_id` to `PassiveCOmmitteeInfo`.